src/psyclone/domain/lfric/kernel_interface.py

```diff
@@ -364,9 +364,14 @@ def operator(self, arg, var_accesses=None):
             symbol_type=LFRicTypes("NumberOfDofsDataSymbol"),
             interface=self._read_access)
 
-        ncells = LFRicTypes("NumberOfCellsDataSymbol")(
-            "ncell_3d", interface=self._read_access)
-        self._symbol_table.add(ncells)
+        try:
+            # We may already have this argument as we add it for each
+            # operator (TODO #2074).
+            ncells = self._symbol_table.lookup("ncell_3d")
+        except KeyError:
+            ncells = LFRicTypes("NumberOfCellsDataSymbol")(
+                "ncell_3d", interface=self._read_access)
+            self._symbol_table.add(ncells)
```

Comment:
I agree, it would be nicer. However, I'm not sure I can do that and still use the `LFRicTypes("NumberOfCellsDataSymbol")`. Perhaps the LFRic-specific symbol table should override `find_or_create` so that it can accept an `lfric_type` argument or similar?

Reply:
Actually, I realised that it's quite possible to use `find_or_create` so I've done that.